specutils/fitting/fitmodels.py

```diff
@@ -212,6 +212,44 @@ def _fit_lines(spectrum, model, fitter=fitting.LevMarLSQFitter(),
     return fit_model
 
 
+def _convert_and_dequantify(poss_quantity, dispersion_unit, dispersion, flux_unit):
+    """
+    This method will convert the ``poss_quantity`` value to the proper
+    dispersion or flux units and then strip the units.
+
+    If the ``poss_quantity`` is None, or a number, we just return that...
+
+    Note: This method can be removed along with most of the others here
+          when astropy.fitting will fit models that contain units.
+
+    """
+
+    if poss_quantity is None or isinstance(poss_quantity, (float, int)):
```

Comment:
You might want to add an `isinstance(poss_quantity, Quantity)` clause here. It's *possible* that in the future someone (or a future version of `astropy,modeling` would see Quantity's coming in directly...

Reply:
Uh, I don't quite understand. For that line we are specifically checking for *not* Quantities. is that what you meant?